Include output truncation message in tool call results

pakrym-oai opened the pull request:
To avoid model being confused about incomplete output.

pakrym-oai committed:
Include truncation message in tool call results

pakrym-oai committed:
codex: improve output truncation message for readability and clarity

pakrym-oai committed:
tests: fix test_exec_stderr_stream_events_echo() to check text fields of stdout/stderr

pakrym-oai committed:
tests/landlock.rs: fix stdout/stderr access after api change to use .text field

pakrym-oai committed:
landlock: use text field from stdout/stderr in assert_network_blocked output handling

dylan-hurd-oai reviewed (approved):
Some small comments, but LGTM

pakrym-oai committed:
core/tests/exec.rs: refactor test setup, error handling, and add skip_test for seatbelt

- Add skip_test helper to skip tests when sandbox var is set.
- Refactor run_test_cmd to return Result and assert expected results in each test.
- Remove should_be_ok param, handle test skipping and assertions in individual tests.
- Simplify output handling and error assertions.

pakrym-oai commented on codex-rs/core/tests/exec.rs:
```diff
@@ -14,10 +15,14 @@ use tokio::sync::Notify;
+    tmp: TempDir,
+    cmd: Vec<&str>,
+    should_be_ok: bool,
+) -> Option<ExecToolCallOutput> {
```
good point, fixed

pakrym-oai committed:
test: allow clippy::unwrap_used in addition to expect_used on macos exec tests
